CI seems go wrong(seems not related with this PR), but i have no idea about that.

```
stanza/operator/input/udp/udp.go:142: File is not `gofmt`-ed with `-s` `-r 'interface{} -> any'` (gofmt)
level=info msg="[runner] processing took 32.17115ms with stages: exclude-rules: 11.786536ms, identifier_marker: 8.651115ms, path_prettifier: 4.33792ms, autogenerated_exclude: 3.482249ms, nolint: 3.399537ms, skip_dirs: 390.69µs, source_code: 48.711µs, cgo: 37.53µs, filename_unadjuster: 19.466µs, max_same_issues: 5.27µs, uniq_by_line: 3.145µs, max_from_linter: 2.225µs, max_per_file_from_linter: 1.694µs, path_prefixer: 1.634µs, path_shortener: 851ns, severity-rules: 561ns, sort_results: 501ns, fixer: 480ns, skip_files: 392ns, exclude: 342ns, diff: 301ns"
level=info msg="[runner] linters took 15.6624762s with stages: goanalysis_metalinter: 15.630148672s"
level=info msg="File cache stats: 229 entries of total size 1.1MiB"
			New: func() interface{} {
level=info msg="Memory: 211 samples, avg is 403.6MB, max is 699.6MB"
level=info msg="Execution took 22.229189021s"
make[2]: *** [../../Makefile.Common:177: lint] Error 1
make[2]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza'
```

--- a/cmd/checkapi/allowlist.txt
+++ b/cmd/checkapi/allowlist.txt
@@ -2,5 +2,4 @@ connector/servicegraphconnector
 extension/encoding
 extension/observer
 processor/servicegraphprocessor
-receiver/journaldreceiver
 receiver/kafkareceiver

--- a/receiver/journaldreceiver/factory.go
+++ b/receiver/journaldreceiver/factory.go
@@ -1,0 +1,13 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package journaldreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/journaldreceiver"
+
+import (
+	"go.opentelemetry.io/collector/receiver"
+)
+
+// NewFactory creates a factory for journald receiver
+func NewFactory() receiver.Factory {
+	return newFactoryAdapter()
+}

--- a/receiver/journaldreceiver/factory_test.go
+++ b/receiver/journaldreceiver/factory_test.go
@@ -1,0 +1,19 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package journaldreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/journaldreceiver"
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/journaldreceiver/internal/metadata"
+)
+
+func TestNewFactory(t *testing.T) {
+	t.Run("NewFactoryCorrectType", func(t *testing.T) {
+		factory := NewFactory()
+		require.EqualValues(t, metadata.Type, factory.Type())
+	})
+}

--- a/receiver/journaldreceiver/journald.go
+++ b/receiver/journaldreceiver/journald.go
@@ -17,8 +17,8 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/journaldreceiver/internal/metadata"
 )
 
-// NewFactory creates a factory for journald receiver
-func NewFactory() receiver.Factory {
+// newFactoryAdapter creates a factory for journald receiver
+func newFactoryAdapter() receiver.Factory {
 	return adapter.NewFactory(ReceiverType{}, metadata.LogsStability)
 }
 

--- a/receiver/journaldreceiver/journald_nonlinux.go
+++ b/receiver/journaldreceiver/journald_nonlinux.go
@@ -19,8 +19,8 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/journaldreceiver/internal/metadata"
 )
 
-// NewFactory creates a dummy factory.
-func NewFactory() receiver.Factory {
+// newFactoryAdapter creates a dummy factory.
+func newFactoryAdapter() receiver.Factory {
 	return receiver.NewFactory(
 		metadata.Type,
 		createDefaultConfig,

--- a/receiver/journaldreceiver/journald_test.go
+++ b/receiver/journaldreceiver/journald_test.go
@@ -28,7 +28,7 @@ import (
 func TestLoadConfig(t *testing.T) {
 	cm, err := confmaptest.LoadConf(filepath.Join("testdata", "config.yaml"))
 	require.NoError(t, err)
-	factory := NewFactory()
+	factory := newFactoryAdapter()
 	cfg := factory.CreateDefaultConfig()
 
 	sub, err := cm.Sub(component.NewIDWithName(metadata.Type, "").String())
@@ -40,7 +40,7 @@ func TestLoadConfig(t *testing.T) {
 
 func TestInputConfigFailure(t *testing.T) {
 	sink := new(consumertest.LogsSink)
-	factory := NewFactory()
+	factory := newFactoryAdapter()
 	badCfg := &JournaldConfig{
 		BaseConfig: adapter.BaseConfig{
 			Operators: []operator.Config{},

